1.Print numbers from 1 to N

In [2]:
n=int(input("Enter a number: "))
for i in range(1,n):
    print(i)

1
2
3
4
5
6
7
8
9


2.sum of first N numbers

In [ ]:
n=int(input("Enter a number: "))
sum=0
for i in range(1,n+1):
    sum+=i
    print(sum)

1
3
6


3.check even or odd

In [73]:
n=int(input("enter a number:"))
if n%2==0:
    print("Even")
else:
    print("Odd")

Odd


4.check positive,negitive or zero

In [52]:
n=int(input("Enter a number: "))
if n>0:
    print("Positive")
elif n<0:
    print("Negative")
else:
    print("Zero")

Positive


5.Find largest of 3 numbers

In [13]:
a=int(input("Enter first number: "))
b=int(input("Enter 2nd number:"))
c=int(input("Enter 3rd number:"))
if a>b and a>c:
    print("a is largest")
elif b>a and b>c:
    print("b is largest")
else:
    print("c is largest")

c is largest


6.check leap year

In [18]:
n=int(input("Enter a number:"))
if n%4==0 and n%100!=0 or n%400==0:
    print("Leap year")
else:
    print("Not a leap year")

Not a leap year


7.Multiplication table

In [15]:
n=int(input("Enter a number: "))
for i in range(1,11):
    print(n,"*",i,"=",n*i)

18 * 1 = 18
18 * 2 = 36
18 * 3 = 54
18 * 4 = 72
18 * 5 = 90
18 * 6 = 108
18 * 7 = 126
18 * 8 = 144
18 * 9 = 162
18 * 10 = 180


8.factorial

In [22]:
n=int(input("enter a number:"))
fact=1
for i in range(1,n+1):
    fact*=i
print(fact)

120


9.count digits in a number

In [3]:
n=int(input("enter a number:"))
count=0
while n!=0:
    n//=10
    count+=1
print(count)

4


10.reverse a number

In [44]:
n=int(input("Enter a number: "))
reverse=0
while n>0:
    temp=n%10
    reverse=reverse*10+temp
    n//=10
print(reverse)

4321


11.sum of digits

In [45]:
n=int(input("enter a number:"))
sum=0
while n>0:
    temp=n%10
    sum+=temp
    n//=10
print(sum)

10


12.check palindrome number

In [53]:
n=int(input("enter a number:"))
rem,rev=0,0
temp=n
while n>0:
    rem=n%10
    rev=rev*10+rem
    n//=10
if rev==temp:
    print("palindrome")
else:
    print("not palindrome")

palindrome


13.factors of a number

In [54]:
n=int(input("Enter a number: "))
for i in range(1,n+1):
    if n%i==0:
        print(i)

1
2
3
4
6
12


14.check prime number

In [67]:
n=int(input("enter a number:"))
if n>1:
 for i in range(2,n):
    if n%i==0:
        print("not prime")
        break
    else:
        print("prime")
else:
    print("not prime")

prime
prime
prime
prime
prime
prime
prime
prime
prime
prime
prime


15.check armstrong number

In [1]:
n=int(input("Enter a number: "))
temp=n
sum=0
while n>0:
    rem=n%10
    sum+=rem**3
    n//=10
if sum==temp:
    print("Armstrong number")
else:
    print("Not an Armstrong number")

Armstrong number


16.find GCD(HCF) of two numbers

In [61]:
a=int(input("Enter first number: "))
b=int(input("Enter 2nd number:"))
while b!=0:
    a,b=b,a%b
print("GCD is",a)

GCD is 6


find strong number 145
1.separate individual digit-1 4 5
2.find factorial    #1!=1,4!=24,5!=120
3.sum of fact      #1+24+120=145
4.condition check  

In [8]:
n=145
m=n
sum=0
while n>0:
    temp=n%10
    fact=1
    while(temp>0):
        fact*=temp
        temp-=1
    sum+=fact
    n//=10
if(sum==m):
    print(m,"strong number")
else:
    print(m,"not strong number")

145 strong number


In [ ]:
for n in range(100,100000):
    m=n
    sum=0
    while n>0:
       temp=n%10
       fact=1
    while(temp>0):
       fact*=temp
       temp-=1
    sum+=fact
    n//=10
if(sum==m):
    print(m,"strong number")

Neon number
find the given number 9 power 2 --81
8+1=9

In [19]:
n=9 
power=9**2
sum=0
while power>0:
    temp=power%10
    sum+=temp
    power//=10
print(sum)

9


153--armstrong number
find length of number 3
separate each number 1,5,3
1 pow len of number--->1pow3+5pow3+3pow3
sum
check condition 

In [ ]:
n=153
sum=0
length=len(str(n))
r=n
while n>0:
    temp=n%10
    sum+=temp**length
    n//=10
if sum==r:
    print(r,"Armstrong number")
else:
    print(r,"Not an Armstrong number")

153 Armstrong number


automorphic number
any value pow 2
n=25
25pow 2--->625
last number n 


In [1]:
t=(1,4,2,2,3,4,5,3)
d={}
for key in t:
    d[key]=d.get(key,0)+1
print(d)

{1: 1, 4: 2, 2: 2, 3: 2, 5: 1}


In [4]:
#unique number of occurences like true or false
arr=[1,4,2,2,3,4,5,3]
d={}
for key in arr:
    d[key]=d.get(key,0)+1
l=list(d.values())
uniq=set(l)
print(uniq)
for i in uniq:
    if l.count(i)>1:
        print("False")
        break
else:
    print("True")


{1, 2}
False


In [12]:
arr=[1,4,2,2,3]
d={}
for key in arr:
    d[key]=d.get(key,0)+1
l=list(d.values())
for i in range(1,len(l)):
    if(l.count(l[i])>1):
        print(False)
print(True)

False
False
True


In [1]:
x="5"
y=2
print(x*y)

55


In [1]:
n=153
sum=0
length=len(str(n))
r=n
while n>0:
    temp=n%10
    sum+=temp**length
    n//=10
if sum==r:
    print("True")
else:
    print("False")

True


single number

In [ ]:
l=[1,1,2,3,3]
for i in range(len(l)):  #0<5
    if (l.count(l[i])==1): #1->2==1,2->1==1
        print(l[i]) #2

2


reverse numbers 4 3 2 1 0

In [1]:
i=int(input())
while i>=0:
    print(i,end=" ")
    i-=1
     

4 3 2 1 0 

sum of digits  --->1+5+3=9

In [1]:
n=153
sum=0
while n>0:
    temp=n%10
    sum=sum+temp
    n=n//10
print(sum)

9


power of 2

In [3]:
n=int(input("Enter a number: "))
while n > 1:
        if n % 2 == 0:
           print(True)
        n //= 2

True
True
True


In [5]:
n=int(input("Enter a number: "))
print(n>0 and (n&(n-1)==0))  

True


power of 3

In [6]:
n=int(input("Enter a number: "))
while n > 1:
        if n % 3 == 0:
            print(True)
        n //= 3
print(n==1)

True
True
True
